refactor!: changed layout of project - redux

MindTooth opened the pull request:
Continued the good work of https://github.com/MindTooth/fish-code/pull/36.  This should be merged somehow.

Tweaked some styles here and there.  Some work to the Makefile, but some targets were broken, left them alone.

Regarding my own comments, I'll see if I can improve the `--dev` part and maybe look into using uvicorn for Flask as well.

tomrtk committed:
refactor!: changed layout of project

- changed layout of project to easier build a package to deploy.
- stopped using `Poetry` for dependency management as it do not work well with nested packages.
- configured setup.cfg for building package.
- fixed issues with file paths after change of layuout.

tomrtk committed:
fix: entry points for core, ui, etc. calling only the packages

tomrtk committed:
Update pyproject.toml

Co-authored-by: Birger J. Nordølum <contact@mindtooth.no>

tomrtk committed:
Update pyproject.toml

Co-authored-by: Birger J. Nordølum <contact@mindtooth.no>

tomrtk committed:
fix: readme mentions git-lfs

tomrtk committed:
fix: path in testfiles

tomrtk committed:
chore: pip need to be of v21.1 or higher

tomrtk committed:
fix: revert setup.cfg entrypoints

R0flcopt3r committed:
build: adds new targets to makefile

Makes the makefile the one stop shop for anything related to building
this code. It will set up virtual environment, install dependencies and
let you easily run the software.

This removes any fear of polluting your system with pip packages by
mistake. It also allows you to rename the virtual environment to
anything that suits you. It is not recommended to rename the virtual
environment to `venv`. I have tested and it works, but produces
warnings.

updates .gitignore to not track files generated by make.

MindTooth committed:
chore(readme): improve venv creation

MindTooth committed:
chore(makefile): fix nits

MindTooth committed:
chore(style): minor nits

R0flcopt3r commented on Makefile:
```diff
@@ -27,33 +27,36 @@ clean-all: clean-venv
-$(venv):
-	virtualenv $(venv)
+
+# Not sure about the below?  Running `make deps` is broken.
```
seems to work fine on my end with your new changes.

MindTooth replied:
I had problems with `pip` struggling with circular dependencies.  I've added the nina target as a dep for the deps target.  Check the last commit.

MindTooth committed:
chore(makefile): add dep to target

This make sures the packages are correctly installed.
Otherwise I got a problem with circular dep loop with
pip.
